<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>pinhole_correction_editor</class>
 <widget class="QWidget" name="pinhole_correction_editor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>815</width>
    <height>310</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Pinhole Correction Editor</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="3" column="0" colspan="2">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="2" column="0" colspan="2">
    <widget class="HiddenBarTabWidget" name="tab_widget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="none_tab">
      <attribute name="title">
       <string>None</string>
      </attribute>
     </widget>
     <widget class="QWidget" name="layer_tab">
      <attribute name="title">
       <string>Layer</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="2">
        <widget class="QLabel" name="layer_pinhole_thickness_label">
         <property name="text">
          <string>Pinhole Thickness</string>
         </property>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="ScientificDoubleSpinBox" name="layer_pinhole_diameter">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>30</height>
          </size>
         </property>
         <property name="keyboardTracking">
          <bool>false</bool>
         </property>
         <property name="suffix">
          <string> μm</string>
         </property>
         <property name="decimals">
          <number>8</number>
         </property>
         <property name="maximum">
          <double>100000000.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
         <property name="value">
          <double>200.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="ScientificDoubleSpinBox" name="layer_thickness">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>30</height>
          </size>
         </property>
         <property name="keyboardTracking">
          <bool>false</bool>
         </property>
         <property name="suffix">
          <string> μm</string>
         </property>
         <property name="decimals">
          <number>8</number>
         </property>
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="maximum">
          <double>10000000.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
         <property name="value">
          <double>5.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="layer_thickness_label">
         <property name="text">
          <string>Layer Thickness</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="layer_type_label">
         <property name="text">
          <string>Layer Type</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QLabel" name="layer_pinhole_diameter_label">
         <property name="text">
          <string>Pinhole Diameter</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="ScientificDoubleSpinBox" name="layer_pinhole_thickness">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>30</height>
          </size>
         </property>
         <property name="keyboardTracking">
          <bool>false</bool>
         </property>
         <property name="suffix">
          <string> μm</string>
         </property>
         <property name="decimals">
          <number>8</number>
         </property>
         <property name="minimum">
          <double>0.100000000000000</double>
         </property>
         <property name="maximum">
          <double>10000000.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
         <property name="value">
          <double>100.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QComboBox" name="layer_type">
         <property name="currentIndex">
          <number>-1</number>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="ScientificDoubleSpinBox" name="layer_standoff">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>30</height>
          </size>
         </property>
         <property name="keyboardTracking">
          <bool>false</bool>
         </property>
         <property name="suffix">
          <string> μm</string>
         </property>
         <property name="decimals">
          <number>8</number>
         </property>
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="maximum">
          <double>10000000.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
         <property name="value">
          <double>150.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="4" column="0" colspan="3">
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>1</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="layer_standoff_label">
         <property name="text">
          <string>Layer Standoff</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="rygg_tab">
      <attribute name="title">
       <string>Pinhole (Rygg)</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="0">
        <widget class="QLabel" name="rygg_diameter_label">
         <property name="text">
          <string>Pinhole Diameter</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="ScientificDoubleSpinBox" name="rygg_absorption_length_value">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The Rygg pinhole correction requires the pinhole's absorption length. Select it from one of several methods:&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;1. &amp;quot;Enter Manually&amp;quot; - type in the absorption length manually&lt;/p&gt;&lt;p&gt;2. Select a default pinhole material. These are after the first separator in the combo box. The beam energy in the current instrument settings will be used to compute an absorption length, and it will be displayed in the disabled spinbox on the right.&lt;/p&gt;&lt;p&gt;3. Select a material from your materials list. These are after the second separator in the combo box. The absorption length will be displayed in the disabled spinbox on the right. If you add materials via the materials panel in the application, they will show up here.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="suffix">
          <string> μm</string>
         </property>
         <property name="decimals">
          <number>8</number>
         </property>
         <property name="minimum">
          <double>0.000001000000000</double>
         </property>
         <property name="maximum">
          <double>100000000.000000000000000</double>
         </property>
         <property name="value">
          <double>100.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>10</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="2" column="1">
        <widget class="QComboBox" name="rygg_absorption_length_selector">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The Rygg pinhole correction requires the pinhole's absorption length. Select it from one of several methods:&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;1. &amp;quot;Enter Manually&amp;quot; - type in the absorption length manually&lt;/p&gt;&lt;p&gt;2. Select a default pinhole material. These are after the first separator in the combo box. The beam energy in the current instrument settings will be used to compute an absorption length, and it will be displayed in the disabled spinbox on the right.&lt;/p&gt;&lt;p&gt;3. Select a material from your materials list. These are after the second separator in the combo box. The absorption length will be displayed in the disabled spinbox on the right. If you add materials via the materials panel in the application, they will show up here.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="rygg_thickness_label">
         <property name="text">
          <string>Pinhole Thickness</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QLabel" name="rygg_num_phi_elements_label">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The number of φ elements to use for integration.&lt;/p&gt;&lt;p&gt;Larger is more accurate, but also takes longer to compute.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Number of φ Elements&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="ScientificDoubleSpinBox" name="rygg_thickness">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>30</height>
          </size>
         </property>
         <property name="keyboardTracking">
          <bool>false</bool>
         </property>
         <property name="suffix">
          <string> μm</string>
         </property>
         <property name="decimals">
          <number>8</number>
         </property>
         <property name="maximum">
          <double>100000000.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
         <property name="value">
          <double>100.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="3" column="0" colspan="3">
        <widget class="QPushButton" name="rygg_apply_panel_buffers">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Apply panel buffers computed via the pinhole dimensions. These will be merged with any other panel buffers already present on the detectors.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>Apply Panel Buffers from Pinhole Dimensions</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QSpinBox" name="rygg_num_phi_elements">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>30</height>
          </size>
         </property>
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The number of φ elements to use for integration.&lt;/p&gt;&lt;p&gt;Larger is more accurate, but also takes longer to compute.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="keyboardTracking">
          <bool>false</bool>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>10000</number>
         </property>
         <property name="value">
          <number>60</number>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="rygg_absorption_length_label">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The Rygg pinhole correction requires the pinhole's absorption length. Select it from one of several methods:&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;1. &amp;quot;Enter Manually&amp;quot; - type in the absorption length manually&lt;/p&gt;&lt;p&gt;2. Select a default pinhole material. These are after the first separator in the combo box. The beam energy in the current instrument settings will be used to compute an absorption length, and it will be displayed in the disabled spinbox on the right.&lt;/p&gt;&lt;p&gt;3. Select a material from your materials list. These are after the second separator in the combo box. The absorption length will be displayed in the disabled spinbox on the right. If you add materials via the materials panel in the application, they will show up here.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>Absorption Length:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="ScientificDoubleSpinBox" name="rygg_diameter">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>30</height>
          </size>
         </property>
         <property name="keyboardTracking">
          <bool>false</bool>
         </property>
         <property name="suffix">
          <string> μm</string>
         </property>
         <property name="decimals">
          <number>8</number>
         </property>
         <property name="maximum">
          <double>100000000.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
         <property name="value">
          <double>200.000000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="jhe_tab">
      <attribute name="title">
       <string>Pinhole (JHE)</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_7">
       <item row="2" column="0" colspan="2">
        <widget class="QPushButton" name="jhe_apply_panel_buffers">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Apply panel buffers computed via the pinhole dimensions. These will be merged with any other panel buffers already present on the detectors.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>Apply Panel Buffers from Pinhole Dimensions</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="jhe_diameter_label">
         <property name="text">
          <string>Pinhole Diameter</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="ScientificDoubleSpinBox" name="jhe_diameter">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>30</height>
          </size>
         </property>
         <property name="keyboardTracking">
          <bool>false</bool>
         </property>
         <property name="suffix">
          <string> μm</string>
         </property>
         <property name="decimals">
          <number>8</number>
         </property>
         <property name="maximum">
          <double>100000000.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
         <property name="value">
          <double>200.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="jhe_thickness_label">
         <property name="text">
          <string>Pinhole Thickness</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="ScientificDoubleSpinBox" name="jhe_thickness">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>30</height>
          </size>
         </property>
         <property name="keyboardTracking">
          <bool>false</bool>
         </property>
         <property name="suffix">
          <string> μm</string>
         </property>
         <property name="decimals">
          <number>8</number>
         </property>
         <property name="maximum">
          <double>100000000.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
         <property name="value">
          <double>100.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="3" column="0" colspan="2">
        <spacer name="verticalSpacer_5">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>10</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QComboBox" name="correction_type">
     <item>
      <property name="text">
       <string>None</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Layer</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Pinhole (Rygg)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Pinhole (JHE)</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QLabel" name="correction_type_label">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Correction Type:</string>
     </property>
    </widget>
   </item>
   <item row="4" column="0" colspan="2">
    <widget class="QCheckBox" name="apply_to_polar_view">
     <property name="toolTip">
      <string>Apply the provided correction settings to the polar view.

This will override any overlays that are currently being used for polar distortion.</string>
     </property>
     <property name="text">
      <string>Apply correction to polar view</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ScientificDoubleSpinBox</class>
   <extends>QDoubleSpinBox</extends>
   <header>scientificspinbox.h</header>
  </customwidget>
  <customwidget>
   <class>HiddenBarTabWidget</class>
   <extends>QTabWidget</extends>
   <header>hidden_bar_tab_widget</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>correction_type</tabstop>
  <tabstop>tab_widget</tabstop>
  <tabstop>layer_pinhole_thickness</tabstop>
  <tabstop>layer_pinhole_diameter</tabstop>
  <tabstop>rygg_diameter</tabstop>
  <tabstop>rygg_thickness</tabstop>
  <tabstop>rygg_num_phi_elements</tabstop>
  <tabstop>rygg_absorption_length_selector</tabstop>
  <tabstop>rygg_absorption_length_value</tabstop>
  <tabstop>rygg_apply_panel_buffers</tabstop>
  <tabstop>jhe_diameter</tabstop>
  <tabstop>jhe_thickness</tabstop>
  <tabstop>jhe_apply_panel_buffers</tabstop>
  <tabstop>apply_to_polar_view</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>correction_type</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>tab_widget</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>476</x>
     <y>192</y>
    </hint>
    <hint type="destinationlabel">
     <x>405</x>
     <y>276</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
